File: day7.py
with open("input7.txt") as f:
    inp = [int(n) for n in f.read().strip().split(",")]

part1, part2 = float("inf"), float("inf")

for pos in range(min(inp), max(inp)):

    diffs = [abs(pos - i) for i in inp]
    sums = [(n * (n + 1)) / 2 for n in diffs]

    part1 = min(part1, sum(diffs))
    part2 = min(part2, sum(sums))

print(part1, int(part2))

File: day7.jl
using Statistics, Printf

inp = readline("input7.txt")
inp = [parse(Int32, i) for i in split(inp,',')]

# part 1
println(sum(abs.(inp.-Int(median(inp)))))

# part 2
cost = n -> n*(n+1)/2
c1 = sum(cost.(abs.(inp.-Int(ceil(mean(inp))))))
c2 = sum(cost.(abs.(inp.-Int(floor(mean(inp))))))
@printf("%d", min(c1,c2))
